src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Connection.cs

```diff
@@ -141,24 +127,10 @@ public void Start()
 
         public Task StopAsync()
         {
-            lock (_stateLock)
-            {
-                switch (_connectionState)
-                {
-                    case ConnectionState.SocketClosed:
-                        return TaskUtilities.CompletedTask;
-                    case ConnectionState.CreatingFrame:
-                        _connectionState = ConnectionState.ToDisconnect;
-                        break;
-                    case ConnectionState.Open:
-                        _frame.Stop();
-                        SocketInput.CompleteAwaiting();
-                        break;
-                }
+            _frame.Stop();
+            _frame.SocketInput.CompleteAwaiting();
```

Comment:
Might actually want to add a test for this. Could be done with a custom connection filter.